packages/core/src/internal/controllers/aria-disabled.controller.ts

```diff
@@ -0,0 +1,26 @@
+import { ReactiveController, ReactiveElement } from 'lit';
+
+export type AriaDisabled = ReactiveElement & { disabled: boolean; readonly?: boolean };
+
+/**
+ * Provides all nessesary aria-* attributes to create valid disabled state for interactive components.
+ */
+export function ariaDisabled<T extends AriaDisabled>(): ClassDecorator {
+  return (target: any) => target.addInitializer((instance: T) => new AriaDisabledController(instance));
+}
+
+export class AriaDisabledController<T extends AriaDisabled> implements ReactiveController {
+  constructor(private host: T) {
+    this.host.addController(this);
+  }
+
+  hostUpdated() {
+    if (this.host.disabled !== null) {
+      this.host.ariaDisabled = this.host.disabled;
+    }
+
+    if (this.host.readonly) {
+      this.host.ariaDisabled = null;
+    }
```

Comment:
When a host component is `readonly` typically this means the component is content and not an interactive element. For example readonly tags display information but are not clickable. In these cases we do not want to describe any aria states and treat it as if it was static content.